test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -675,7 +675,7 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
         }
     }
 
-    class ExtensionState implements ExtensionContext.Store.CloseableResource {
+    class ExtensionState {
```

Comment:
By removing the interface, we ensure that JUnit does not attempt to close this, because we are closing it ourselved

Reply:
How is this being shutdown then? By the shutdown hook?

Reply:
Yes. When we let Junit 5 shut it down, there are a few problems:

1) It gets shutdown multiple times (which can make `QuarkusTestResourceLifecycleManager` fail - I saw this when looking at the Surefire M5 bump PR)
2) JUnit 5 shuts it down from the wrong ClassLoader. This is what @pedroigor found yesterday when debugging the mysterious failures I was seeing for the Keycloak tests